shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/login/LoginViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package co.nimblehq.blisskmmic.presentation.modules.login
+
+import co.nimblehq.blisskmmic.data.network.helpers.jsonApiException
+import co.nimblehq.blisskmmic.domain.model.Token
+import co.nimblehq.blisskmmic.domain.usecase.LogInUseCase
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+data class LoginViewState(
+    val isSuccess: Boolean = false,
+    val isLoading: Boolean = false,
+    val error: String? = null
+) {
+    @Suppress("EmptySecondaryConstructor")
+    constructor() : this(false, false, null) {}
+
+    @Suppress("EmptySecondaryConstructor")
+    constructor(error: String?) : this(false, false, error) {}
+}
+
+class LoginViewModel(private val logInUseCase: LogInUseCase) {
+
+    private val mutableViewState: MutableStateFlow<LoginViewState> =
+        MutableStateFlow(LoginViewState())
+    private val viewScope = MainScope()
```

Comment:
The API request should be executed on `io` scope instead 🙏

Reply:
@luongvo  Could you advice alternative or how to. I found this [conversation](https://github.com/Kotlin/kotlinx.coroutines/issues/3205) that saying that using `io` scope is not yet possible. 🙏

Reply:
Use ViewModelScope in Android and MainScope for iOS https://github.com/blyscuit/kmm-swiftUI-IC/pull/53/commits/f17665c839bc60aef87e7ede0c98c43947bc8bae.